tests/storage_test.go

```diff
@@ -306,5 +308,68 @@ var _ = Describe("Storage", func() {
 				}
 			})
 		})
+
+		Context("With a HostDisk defined", func() {
+			table.DescribeTable("should be sucessfully started", func(hostDiskType v1.HostDiskType) {
+				By("starting VirtualMachineInstance")
+				vmi := tests.NewRandomVMIWithHostDisk(hostDiskType)
+				RunVMIAndExpectLaunch(vmi, false, 10)
+
+				By("checking if disk.img exists")
+				vmiPod := tests.GetRunningPodByVirtualMachineInstance(vmi, tests.NamespaceTestDefault)
+				output, err := tests.ExecuteCommandOnPod(
+					virtClient,
+					vmiPod,
+					vmiPod.Spec.Containers[0].Name,
+					[]string{"find", "/data", "-name", "disk.img", "-size", "1G"},
+				)
+				Expect(strings.Contains(output, "/data/disk.img")).To(BeTrue())
```

Comment:
Since the path is hardcoded I would expect that after an initial test run the disk.img already exists and subsequent runs would not test exactly the same scenario, is that the case?

Reply:
Yes, that's the case here, I am testing ```DiskExistsOrCreate``` (first run should create a disk.img) and then ```DiskExists``` (the second run should not receive any error since disk.img has been created)

Reply:
That would mean that if I run `make functest` two times, I am testing different things. We should not do that.

Reply:
ofc.. you're right, don't know why I expected every ```make functest``` will run on clean cluster..
I will fix that. Thanks.

Reply:
Btw. also expect that tests can be executed in a randomized order (there is an extra ginkgo flag for that).

Reply:
I see different ways:

 * splitting the suite into two contexts, one where a disk is already created in `BeforeEach` or `BeforeAll` and one where it is not
 * Just flattening the table test in this case and describe with `By()` what you do

Not sure if a loop helps you a lot when you want do document the tests

To test the case where the image already exists in a pvc, using the alpine pvc is definitely good. 

In general you could also use random image names per `It` if they should not overlap and simply delete all images in that folder in a `BeforeSuite` (to make it reentrant in error cases) and `AfterSuite`.

Reply:
Done.